File: 150 interview ques/Q1.py
"""
https://leetcode.com/problems/merge-sorted-array/?envType=study-plan-v2&envId=top-interview-150
"""
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        
        a = m - 1  
        b = n - 1  
        c = m + n - 1  
                           
        
        while a >= 0 and b >= 0:
            if nums1[a] > nums2[b]:
                nums1[c] = nums1[a]  
                a -= 1  
            else:
                nums1[c] = nums2[b]  
                b -= 1  
            c -= 1  

                    
        while b >= 0:
           nums1[c] = nums2[b]
           b -= 1
           c -= 1

File: 150 interview ques/Q1.c
/*https://leetcode.com/problems/two-sum/description/?envType=study-plan-v2&envId=top-interview-150*/
#include<stdio.h>
int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    int i, j;
    for (i = 0; i < numsSize; i++) {
        for (j = i + 1; j < numsSize; j++) {
            if (nums[i] + nums[j] == target) {
                int* result = (int*)malloc(2 * sizeof(int)); // Allocate memory
                result[0] = i;
                result[1] = j;
                *returnSize = 2; // Set return size
                return result;
            }
        }
    }
    *returnSize = 0; // No result found
    return NULL;
}
